tests/chainer_tests/links_tests/normalization_tests/test_decorrelated_batch_normalization.py

```diff
@@ -0,0 +1,139 @@
+    'dtype': [numpy.float32, numpy.float64],
+})))
+class BatchNormalizationTest(unittest.TestCase):
+
+    def setUp(self):
+        C = self.n_channels // self.groups
+        head_ndim = 2
+
+        self.link = links.DecorrelatedBatchNormalization(self.n_channels,
+                                                         groups=self.groups,
+                                                         dtype=self.dtype)
+        self.link.cleargrads()
+
+        shape = (5, self.n_channels) + (2,) * self.ndim
+        self.x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
+        self.gy = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
+
+        if self.test:
+            self.mean = numpy.random.uniform(-1, 1, (C,)).astype(self.dtype)
+            self.projection = numpy.random.uniform(0.5, 1, (C, C)).astype(
+                self.dtype)
+            self.link.expected_mean[...] = self.mean
+            self.link.expected_projection[...] = self.projection
+        else:
+            spatial_axis = tuple(range(head_ndim, self.x.ndim))
+            x_hat = self.x.reshape((5 * self.groups, C, ) + self.x.shape[2:])
+            x_hat = x_hat.transpose((1, 0) + spatial_axis).reshape((C, -1))
+            self.mean = x_hat.mean(axis=1)
+            self.projection = _calc_projection(self.x, self.mean,
+                                               self.link.eps, self.groups)
+        self.check_forward_options = {'atol': 1e-4, 'rtol': 1e-3}
+        self.check_backward_options = {'atol': 1e-4, 'rtol': 1e-3, 'dtype':
+                                       numpy.float64}
+        if self.dtype == numpy.float16:
+            self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
+            self.check_backward_options = {'atol': 5e-1, 'rtol': 1e-1}
+        elif self.dtype == numpy.float32:
+            self.check_backward_options = {
+                'dtype': numpy.float64, 'atol': 1e-2, 'rtol': 1e-2}
+
+    def check_forward(self, x_data):
+        with chainer.using_config('train', not self.test):
+            x = chainer.Variable(x_data)
+            y = self.link(x)
+            self.assertEqual(y.data.dtype, self.dtype)
+
+        y_expect = _decorrelated_batch_normalization((
+            self.x, self.mean, self.projection, self.groups))
+
+        testing.assert_allclose(
+            y_expect, y.data, **self.check_forward_options)
+
+    @condition.retry(3)
+    def test_forward_cpu(self):
+        self.check_forward(self.x)
+
+    @attr.gpu
+    @condition.retry(3)
+    def test_forward_gpu(self):
+        self.link.to_gpu()
+        self.check_forward(cuda.to_gpu(self.x))
+
+    def check_backward(self, x_data, y_grad):
+        gradient_check.check_backward(
+            self.link, x_data, y_grad, (),
+            eps=1e-2, **self.check_backward_options)
+
+    @condition.retry(3)
+    def test_backward_cpu(self):
+        self.check_backward(self.x, self.gy)
+
+    @attr.gpu
+    @condition.retry(3)
```

Comment:
Is it possible to adjust tolerances to avoid relying on `retry` ? Preferably (we do this internally), you'd like to run the test for a couple of thousand repetitions without a single failure.